### ISSUE TITLE
Merging PDF Files destroys external links

### ISSUE THREAD

**Issue by MatthiasKuehneEllerhold** [mpdf/mpdf#900]
### I found this bug
Merging multiple PDF files into one PDF using the snippet below destroys all links (internal and external).

### This is mPDF and PHP version and environment (server/fpm/cli etc) I am using
* PHP 7.2.10
* Mpdf 7.1.6

### This is a PHP code snippet I use

```
<?php
$mpdf->SetImportUse();
foreach ($pdfDocuments as $pdfDocument) {
    $tempName  = tempnam($tempDir, 'merge');
    file_put_contents($tempName, $pdfDocument);
    $pageCount = $mpdf->SetSourceFile($tempName);

    for ($page = 1; $page <= $pageCount; $page++) {
        $mpdf->AddPage();
        $template = $mpdf->ImportPage($page);
        $mpdf->UseTemplate($template);
    }

    unlink($tempName);
}

$mpdf->SetTitle($title);

return $mpdf->Output('', 'S');
```

### This is a HTML/CSS code snippet I use

```
<html>
<body>
<a href="https://www.example.org">This ia a test link</a>
</body>
</html>
```

A solution for FPDI is offered here: https://gist.github.com/andreyvit/2020422#gistcomment-317454
I've adapted it for mpdf and it works. Do you want me to provide a PR or will this be fixed with fpdi 2.x automatically?

**Comment by jakejackson1** [comment]
As far as I'm aware, FPDI 2.x also doesn't include the annotations / hyperlinks so a PR [will need to be made for 2.x once it lands](https://github.com/mpdf/mpdf/pull/899).

**Comment by MatthiasKuehneEllerhold** [comment]
I've added a PR. Two remarks:

1. Im unsure about the Code Style used. My code was copy and pasted from PSR2 - which is entirely contrary to the whole Mpdf class. 

2. In order to preserve the hyperlinks a new method `getPages` in `fpdi_pdf_parser` is needed. I've solved it by creating a new subclass `Mpdf\FpdiPdfParser`, but thats not really nice. 
Is there any other way to get all pages of a template?
